java/server/src/org/openqa/grid/internal/utils/configuration/GridHubConfiguration.java

```diff
@@ -0,0 +1,177 @@
+
+  @Parameter(
+    names = "-newSessionWaitTimeout",
+    description = "<Integer> milliseconds. Default to no timeout ( -1 ) the time in ms after which a new test waiting for a node to become available will time out.When that happens, the test will throw an exception before starting a browser."
+  )
+  public Integer newSessionWaitTimeout = -1;
+
+  @Parameter(
+    names = "-prioritizer",
+    description = "a class implementing the Prioritizer interface. Default to null ( no priority = FIFO ).Specify a custom prioritizer if you want to sort the order new session requests are processed when there is a queue.",
+    converter = PrioritizerString.class
+  )
+  public Prioritizer prioritizer = null;
+
+  @Parameter(
+    names = "-throwOnCapabilityNotPresent",
+    description = "<true | false> default to true. If true, the hub will reject test requests right away if no proxy is currently registered that can host that capability.Set it to false to have the request queued until a node supporting the capability is added to the grid."
+  )
+  public Boolean throwOnCapabilityNotPresent = true;
+
+  private static final GridHubConfiguration DEFAULT_CONFIG = loadFromJSON(JSONConfigurationUtils.loadJSON("defaults/DefaultHub.json"));
+
+  public GridHubConfiguration() {
+    if (DEFAULT_CONFIG != null) {
+      merge(DEFAULT_CONFIG);
+    }
+  }
+
+  /**
+   * @param json JsonObject to load configuration from
+   */
+  public static GridHubConfiguration loadFromJSON(JsonObject json) {
+
+    try {
+      return new JsonToBeanConverter().convert(GridHubConfiguration.class, json);
+    } catch (Throwable e) {
+      throw new GridConfigurationException("Error with the JSON of the config : " + e.getMessage(),
+                                           e);
+    }
+  }
+
+  private class PrioritizerString implements IStringConverter<Prioritizer> {
+    @Override
+    public Prioritizer convert(String prioritizerClass) {
+      try {
+        return (Prioritizer) Class.forName(prioritizerClass).newInstance();
+      } catch (Throwable e) {
+        throw new GridConfigurationException("Error creating the prioritize from class " +
+                                             prioritizerClass + " : " + e.getMessage(), e);
+      }
+    }
+  }
+
+  private class CapabilityMatcherString implements IStringConverter<CapabilityMatcher> {
+    @Override
+    public CapabilityMatcher convert(String capabilityMatcherClass) {
+      try {
+        return (CapabilityMatcher) Class.forName(capabilityMatcherClass).newInstance();
+      } catch (Throwable e) {
+        throw new GridConfigurationException("Error creating the prioritize from class " +
+                                             capabilityMatcherClass + " : " + e.getMessage(), e);
+      }
+    }
+  }
+
+
+  public void merge(GridNodeConfiguration other) {
+    super.merge(other);
+  }
+
+  public void merge(GridHubConfiguration other) {
```

Comment:
the super classes have methods that define their concrete class name, no Override here (IDE complains if you do add it, because it's not overriding the same signature)